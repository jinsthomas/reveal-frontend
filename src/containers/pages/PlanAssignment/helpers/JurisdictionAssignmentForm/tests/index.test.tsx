import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import MockDate from 'mockdate';
import React from 'react';
import { plans } from '../../../../../../store/ducks/opensrp/PlanDefinition/tests/fixtures';
import { JurisdictionAssignmentForm } from '../index';
import { openSRPJurisdiction } from './fixtures';

/* tslint:disable-next-line no-var-requires */
// const fetch = require('jest-fetch-mock');

describe('PlanAssignment/JurisdictionAssignmentForm', () => {
  beforeEach(() => {
    jest.resetAllMocks();
    MockDate.set('12/30/2019', 0);
  });

  it('renders without crashing', () => {
    const props = {
      defaultValue: [{ label: 'Team X', value: 'x' }],
      jurisdiction: openSRPJurisdiction,
      options: [
        { label: 'Team X', value: 'x' },
        { label: 'Team Y', value: 'y' },
      ],
      plan: plans[1],
    };

    const wrapper = mount(<JurisdictionAssignmentForm {...props} />);

    expect(wrapper.find('Form').length).toEqual(1);
    expect(wrapper.find('SelectField').props() as any).toMatchSnapshot({
      form: expect.any(Object),
    });
    expect(toJson(wrapper.find('button'))).toMatchSnapshot('Submit buttons');
    wrapper.unmount();
  });

  it('shows error if no plan or jurisdiction', () => {
    const props = {
      defaultValue: [{ label: 'Team X', value: 'x' }],
      options: [
        { label: 'Team X', value: 'x' },
        { label: 'Team Y', value: 'y' },
      ],
    };

    const wrapper = mount(<JurisdictionAssignmentForm {...props} />);

    expect(toJson(wrapper.find('span'))).toMatchSnapshot('no plan or jurisdiction');
    wrapper.unmount();
  });
});
