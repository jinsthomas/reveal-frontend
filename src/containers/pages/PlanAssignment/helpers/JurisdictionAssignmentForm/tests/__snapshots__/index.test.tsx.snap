// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanAssignment/JurisdictionAssignmentForm renders without crashing 1`] = `
Object {
  "children": undefined,
  "className": "form-control",
  "defaultValue": Array [
    Object {
      "label": "Team X",
      "value": "x",
    },
  ],
  "field": Object {
    "name": "organizations",
    "onBlur": [Function],
    "onChange": [Function],
    "value": Array [
      "x",
    ],
  },
  "form": Any<Object>,
  "options": Array [
    Object {
      "label": "Team X",
      "value": "x",
    },
    Object {
      "label": "Team Y",
      "value": "y",
    },
  ],
}
`;

exports[`PlanAssignment/JurisdictionAssignmentForm renders without crashing: Submit buttons 1`] = `
Array [
  <button
    aria-label={null}
    className="btn-block btn btn-secondary btn-sm"
    onClick={[Function]}
    type="button"
  >
    Close
  </button>,
  <button
    aria-label={null}
    className="btn-block btn btn-primary btn-sm"
    disabled={false}
    onClick={[Function]}
    type="submit"
  >
    Save
  </button>,
]
`;

exports[`PlanAssignment/JurisdictionAssignmentForm shows error if no plan or jurisdiction: no plan or jurisdiction 1`] = `
<span>
  Could not load the form.
</span>
`;
